src/components/marquee-wizard/marquee-wizard.component.spec.ts

```diff
@@ -273,3 +282,137 @@ describe('Marquee wizard with delayed step creation', () => {
+class MarqueeWizardVisitedChangeTestComponent {
+    steps: WizardStep[] = [
+        {
+            header: '1. First Step',
+            content: 'Content of step 1.',
+        },
+        {
+            header: '2. Second Step',
+            content: 'Content of step 2.',
+        },
+        {
+            header: '3. Third Step',
+            content: 'Content of step 3.',
+        },
+        {
+            header: '4. Fourth Step',
+            content: 'Content of step 4.',
+        }
+    ];
+    stepChanging(_: StepChangingEvent) {}
+    stepChange(_: number) {}
+    onNext(_: number) {}
+    visitedChanged(_: boolean) { }
+
+    @ViewChildren(MarqueeWizardStepComponent)
+    stepsList: QueryList<MarqueeWizardStepComponent>;
+}
+
+describe('Marquee wizard with visitedChange event', () => {
+    let component: MarqueeWizardVisitedChangeTestComponent;
+    let fixture: ComponentFixture<MarqueeWizardVisitedChangeTestComponent>;
+    let nativeElement: HTMLElement;
+    let visitedChanged: jasmine.Spy;
+
+    beforeEach(async(() => {
+        TestBed.configureTestingModule({
+            imports: [MarqueeWizardModule, CommonModule],
+            declarations: [MarqueeWizardVisitedChangeTestComponent]
+        }).compileComponents();
+    }));
+
+    beforeEach(async () => {
+        fixture = TestBed.createComponent(MarqueeWizardVisitedChangeTestComponent);
+        component = fixture.componentInstance;
+        nativeElement = fixture.nativeElement;
+        visitedChanged = spyOn(component, 'visitedChanged');
+
+        fixture.detectChanges();
+        await fixture.whenStable();
+        fixture.detectChanges();
+        await fixture.whenStable();
+    });
+
+    it('should trigger a visitedChange event when valid modified on the current step, when other steps ahead are visited', async () => {
+        // set step 1 to valid and move forward
+        await clickButton(MarqueeWizardSelectors.ToggleValidity);
+        await clickButton(MarqueeWizardSelectors.NextButton);
+
+        // set step 2 to valid and move forward
+        await clickButton(MarqueeWizardSelectors.ToggleValidity);
+        await clickButton(MarqueeWizardSelectors.NextButton);
+
+        // set step 3 to valid and move forward
+        await clickButton(MarqueeWizardSelectors.ToggleValidity);
+        await clickButton(MarqueeWizardSelectors.NextButton);
+
+        // jump back to the first step
+        await clickButton(MarqueeWizardSelectors.SecondStep);
+
+        visitedChanged.calls.reset();
+
+        // valid now false and should trigger visitedChange
+        await clickButton(MarqueeWizardSelectors.ToggleValidity);
+
+        await clickButton(MarqueeWizardSelectors.NextButton);
+
+        expect(visitedChanged).toHaveBeenCalledWith(false);
+        expect(visitedChanged).toHaveBeenCalledTimes(1);
+
+        const stepsList = component.stepsList.toArray();
+        // step 1 should be valid and visited
+        expect(stepsList[0]._valid).toBeTruthy();
```

Comment:
Can we use the `valid` property instead as we want to test the public API, not the private stuff. We do this for multiple reasons:

1) We want to use the same properties as the consuming application would be using to make sure they work
2) It can help prevent breaking changes, if our tests are relying on the public properties, if we were to remove them or change their data type (ie. breaking change) then the test will throw compilation errors
3) Private APIs are allowed to be changed at any time, apps shouldn't be using them so we can making breaking changes on them any time without affecting users. By relying on private APIs in the tests this means anytime we change the private implementation we will have to update the tests, which in turn means are tests are likely not going to catch any bugs introduced by the changes.